sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -385,7 +385,8 @@ def __init__(
 
   def available(self):
     # pylint: disable=subprocess-run-check
-    trial = subprocess.run(['which', 'java'], capture_output=True)
+    trial = subprocess.run(['which', subprocess_server.JavaHelper.get_java()],
```

Comment:
there is an edge case where only JAVA_HOME is provided. The external transform expansion should work, but here it assumes java not available